en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/defining-interceptors-in-an-open-api-definition.md

```diff
@@ -0,0 +1,244 @@
+    you define your own request body to be sent to the backend. Even though you do not read the request body, the above inclusion
+    should be done to update the request body.
+
+## Response flow Interceptor
+
+You can use the extension `x-wso2-response-interceptor` to define the interceptor for response flow.
+It is same as the request flow, but additionally following values are supported in the `includes` section of the extension.
+
+```yaml
+includes:
+ - response_headers
+ - response_body
+ - response_trailers
+```
+
+``` yaml tab="Format"
+x-wso2-response-interceptor:
+  scheme: <one of [http|https]>
+  host: <host>
+  port: <port>
+  includes: # any of following
+    - request_headers
+    - request_body
+    - request_trailers
+    - response_headers
+    - response_body
+    - response_trailers
+    - invocation_context
+  requestTimeout: OPTIONAL <request timeout in seconds>
+  clusterTimeout: OPTIONAL <override interceptor cluster timeout in seconds>
+```
+
+``` yaml tab="Example API Level"
+.
+.
+info:
+  version: 1.0.5
+  title: PizzaShackAPI
+x-wso2-basePath: /v2
+x-wso2-production-endpoints:
+    urls:
+    - https://localhost:2380/v2
+x-wso2-response-interceptor:
+  scheme: http
+  host: host.request.interceptor
+  port: 9081
+  includes:
+    - request_headers
+    - request_body
+    - response_headers
+.
+.
+```
+
+``` yaml tab="Example Resource Level"
+.
+.
+info:
+  version: 1.0.5
+  title: PizzaShackAPI
+x-wso2-basePath: /v2
+x-wso2-production-endpoints:
+    urls:
+    - https://localhost:2380/v2
+paths:
+    /pet/findByStatus:
+        x-wso2-production-endpoints:
+            urls:
+            -  https://localhost:2380/v1
+        x-wso2-response-interceptor:
+            scheme: http
+            host: host.request.interceptor
+            port: 9081
+            includes:
+            - request_body
+            - invocation_context
+        get:
+            tags:
+            - pets
+            summary: Finds Pets by status
+            description: Multiple status values can be provided with comma separated strings
+            operationId: findPetsByStatus
+            parameters:
+            - name: status
+              in: query
+              description: Status values that need to be considered for filter
+        .
+        .
+    /pet/{petId}:
+        get:
+.
+.
+```
+
+<!-- The content of the below warning is same as the info notice in the file
+deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/response-flow-interceptor.md -->
+!!! info
+    If you update the response body before reaching the client, make sure to add `response_body` to the `includes` section
+    of the Open API Specification. You may have a scenario like, what ever the response body from the backend, based on a header value
+    you define your own response body to be sent to the client. Even though you do not read the response body, the above inclusion
```

Comment:
We can update this too into something similar to https://github.com/wso2/docs-apim/pull/4953/files#r740149614